kernel/sched/fair.c

```diff
@@ -5486,12 +5486,12 @@ static int __assign_cfs_rq_runtime(struct cfs_bandwidth *cfs_b,
 		struct cfs_rq *temp_cfs_rq = (struct cfs_rq *) entry->cfs_rq_p;
 
 		/* Find 99P of yeild and runtime for each runqueue maybe add it back to the queue itself */
-		if (temp_cfs_rq->pa_hist_idx - 1 <= 0)
+		if (temp_cfs_rq->pa_hist_idx<= 0)
 			continue;
-		sort(temp_cfs_rq->pa_yield_time_hist, temp_cfs_rq->pa_hist_idx - 1, sizeof(u64), cmp_u64, NULL);
-		sort(temp_cfs_rq->pa_runtime_hist, temp_cfs_rq->pa_hist_idx - 1, sizeof(u64), cmp_u64, NULL);
+		sort(temp_cfs_rq->pa_yield_time_hist, temp_cfs_rq->pa_hist_idx, sizeof(u64), cmp_u64, NULL);
+		sort(temp_cfs_rq->pa_runtime_hist, temp_cfs_rq->pa_hist_idx, sizeof(u64), cmp_u64, NULL);
 
-		percentile_idx = DIV_ROUND_UP(99 * (temp_cfs_rq->pa_hist_idx - 2), 100);
```

Comment:
-2 here is intentional. P99 wants to remove outliers. for smaller history values, since we round up - P99 will always be the last entry and if there are outliers. On larger history values this also works because we have enough entries for P99 to work anyways.